### PR TITLE
drivers/spi/spi_bitbang: Add private data on spi_bitbang

### PR DESCRIPTION
**Pull request by SPRESENSE**
## Summary
Add private data for spi_bitbang low-level driver to add an instance specific private data.
And add spi_destroy_bitbang() to clean up a driver instance.

## Impact
only spi_bitbang

## Testing
local test with spresense board

### CODE REVIEW

File: drivers/spi/spi_bitbang.c [hunk 1]
```diff
@@ -551,4 +553,12 @@ FAR struct spi_dev_s *spi_create_bitbang(FAR const struct
   return &priv->dev;
 }
 
+void spi_destroy_bitbang(FAR struct spi_dev_s *dev)
+{
+  if (dev)
```
Comment (pkarashchenko) [inline drivers/spi/spi_bitbang.c:558]: Shouldn't this be a DEBUGASSERT(dev);?

**Review by xiaoxiang781216 — APPROVED** [review]

File: drivers/spi/spi_bitbang.c (hunk 1 above)
Comment (pkarashchenko) [reply drivers/spi/spi_bitbang.c:558]: @xiaoxiang781216 why you resolved unanswered comment?

File: include/nuttx/spi/spi_bitbang.h [hunk 2]
```diff
@@ -88,6 +88,7 @@ struct spi_bitbang_s
   uint32_t         holdtime;               /* SCK hold time to achieve requested frequency */
   bitexchange_t    exchange;               /* The select bit exchange function */
   mutex_t          lock;                   /* Supports mutually exclusive access to SPI */
+  FAR void        *priv;                   /* Private data for instance specific */
```
Comment (xiaoxiang781216) [inline include/nuttx/spi/spi_bitbang.h:91]: who use this new field

File: drivers/spi/spi_bitbang.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/spi/spi_bitbang.c:558]: Sorry, mistake.

Comment (hiragane-mitty) [reply drivers/spi/spi_bitbang.c]: @pkarashchenko Fixed it to DEBUGASSERT.

File: include/nuttx/spi/spi_bitbang.h (hunk 2 above)
Comment (hiragane-mitty) [reply include/nuttx/spi/spi_bitbang.h:91]: @xiaoxiang781216 
Anyone can use it. bitbang is kind of SPI upper driver, so lower driver can keep any data for the lower.

Comment (xiaoxiang781216) [reply include/nuttx/spi/spi_bitbang.h:91]: Ok

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Comment by davids5** [comment]
@SPRESENSE Please add to the summary what is this for?

**Pull request merged**